general/v0.1/registration.md

```diff
@@ -27,112 +27,101 @@ administrator. This MAY include but is not limited to the following:
-administrator a registration token that can be copied into the fediverse server configuration.
+A successful registration results in the FASP creating an Ed25519
+keypair and a client identifier (ID) for the fediverse server.
 
-The registration token is a JSON Web Token (JWT) that MUST include the
-registered claims `iss` and `sub` and the private claim `sec`. It MAY
-contain the registered claim `exp`.
+After registration, FASP MUST make an HTTP `POST` request to the
+fediverse server's `/registration` endpoint.
 
-These claims MUST contain the following:
+The payload of that request is a JSON object with the following keys and
+values:
 
-* `iss`: The base URI for API requests to the FASP.
-* `sub`: The OAuth 2.0 client identifier
-* `sec`: The OAuth 2.0 client secret
-* `exp`: An optional expiration time
+* `name`: The name of the FASP - this MUST have been presented to the
+  administrator during registration to make it recognizable.
+* `baseUrl`: The base URL of the FASP
+* `clientId`: The client ID
+* `publicKey`: The public key of the FASP
 
 An example payload:
 
 ```
 {
-  "iss": "https://fasp.example.com",
-  "sub": "b2ks6vm8p23w",
-  "sec": "H4bo9pER2Ww4MlPs2Rf",
-  "exp": 1726498179
+  "name": "Example FASP",
+  "baseUrl": "https://fasp.example.com",
+  "clientId": "b2ks6vm8p23w",
+  "publicKey": "FbUJDVCftINc9FlgRu2jLagCVvOa7I2Myw8aidvkong=" 
 }
 ```
 
-FASPs SHOULD offer a way to re-generate the OAuth credentials and
-this token in case it gets lost or a fediverse server needs to be reinstalled
-from scratch.
+As a result the fediverse server MUST persist this information as a
+request for FASP registration and generate a client ID for the FASP and
+its own Ed25519 keypair for authenticating with the FASP. It MUST then
+reply with an HTTP status code `201` (Created) and a JSON object that
+contains the following keys and values:
 
-The following is a sketch of how this may look in the abstract:
-
-Step 1: A fediverse server admin is presented with a registration form
-
-![A bare-bones sign-up form asking for email, instance URL and acceptance of terms of service](../../images/instance_sign_up.svg)
-
-Step 2: Upon successful registration, the registration token is
-displayed
+* `clientId`: The client ID
+* `publicKey`: The public key of the fediverse server
+* `redirectUri`: An URI to redirect to in order to finish the
+  registration
 
-![A webpage displaying a registration token with a button to copy to clipboard](../../images/instance_sign_up_success.svg)
-
-## Adding a FASP to a Fediverse Server
-
-The option to add a new FASP SHOULD be part of the user interface
-for administrative settings that a fediverse software already has.
+An example payload:
 
-To add a new FASP, the fediverse server adminstrator MUST enter the
-registration token obtained in the previous step. The fediverse
-software MUST validate the token before proceeding.
+```json
+{
+  "clientId": "dfkl3msw6ps3",
```

Comment:
This `clientId` deviates from the FASP `clientId` above — I believe these should be the same value?

Reply:
No, as explained above. I hope the new names make this more clear.